lib/rules/complexity.js

```diff
@@ -109,6 +109,8 @@ module.exports = {
             IfStatement: increaseComplexity,
             WhileStatement: increaseComplexity,
             DoWhileStatement: increaseComplexity,
+            ChainExpression: increaseComplexity,
```

Comment:
This doesn't report the first example from https://github.com/eslint/eslint/issues/18060:

```js
/*eslint complexity: ["error", 2]*/

// still valid (calculated complexity = 2)
function a(x) {
    return x?.a?.b?.c?.d?.e?.f;
}

```

Because this is one `ChainExpression`. Perhaps we should count each `?.`?

Reply:
You are right. I’ve misunderstood how `ChainExpression` works. I’ve added a test-case and updated the implementation. Thank you!